docs/Makefile

```diff
@@ -18,11 +18,21 @@
 # under the License.
 #
 
+ifeq ($(OS),Windows_NT)
+    T3C_DIR_SEP := \\
+	RM := del /s /q
+	PY := python
+else
+    T3C_DIR_SEP := /
+	RM := rm -rf
```

Comment:
RM is just a variable name. It can be anything like, REM or RMM. Based on the OS, It gets substituted with `del /s /q` or `rm -rf`. This is added because default `make clean` fails in windows OS as the `rm -rf `command is not recognized.

Reply:
Right, but you don't need to define it because [`RM` is an "implicit variable"](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html#index-RM)

Reply:
I tried using the RM directly,
```
clean:
	-RM $(BUILDDIR) $(T3C_RST_OUT)
```

But make clean will fail because we are trying to use `rm -f` to delete directories and other files. 

``` make clean                                                             ─╯
RM build source/admin/t3c//t3c-apply.rst source/admin/t3c//t3c-check-refs.rst source/admin/t3c//t3c-check-reload.rst source/admin/t3c//t3c-check.rst source/admin/t3c//t3c-diff.rst source/admin/t3c//t3c-generate.rst source/admin/t3c//t3c-preprocess.rst source/admin/t3c//t3c-request.rst source/admin/t3c//t3c-tail.rst source/admin/t3c//t3c-update.rst source/admin/t3c//t3c.rst source/admin/t3c//testing.rst
RM: build: is a directory
make: [clean] Error 1 (ignored)
````

Hence in earlier version, It was 

```
clean:
	-rm -rf $(BUILDDIR) $(T3C_RST_OUT)
```

And this worked fine with Mac/Linux but not windows. Hence we are using a if else block to assign the variable appropriate meaning. 

To avoid any conflict with in-build functionality i have now renamed the variable from RM to REM.

Reply:
The build directory is its own file, `make` wasn't really meant to make directories and that shows here.
Ideally it'd be that easy, but you can always just do it in two steps
```makefile
clean:
	-$(RM) $(BUILDDIR)/* $(T3C_RST_OUT)
	-rmdir $(BUILDDIR)
```
Since `rmdir` exists on all OSes of which I'm aware, that should always work provided the directory is empty. Keeps you from having to define some variable based on OS - although it looks like you gotta do that anyway with Python for some reason?

Honestly, though, you could also just get rid of the `clean` target. It's not really necessary.

You can do it however you want, but if you want to define a new variable for it, definitely stick with e.g. `REM` rather than re-defining a built-in like `RM`.